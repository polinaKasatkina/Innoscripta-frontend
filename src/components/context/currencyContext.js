import React, { Component } from "react";

const CurrencyContext = React.createContext();

class CurrencyProvider extends Component {

    constructor(props) {
        super(props);

        this.state = {
            currency: "USD",
            usdToEuro : 0
        };

        this.toggleCurrency = this.toggleCurrency.bind(this);
        this.getCurrencyCourse = this.getCurrencyCourse.bind(this);
    }

    componentDidMount() {

        this.getCurrencyCourse();

    }

    toggleCurrency() {
        this.setState(prevState => {
            return {
                currency: prevState.currency === "USD" ? "EUR" : "USD"
            };
        });
    }

    getCurrencyCourse() {


        fetch('https://api.exchangeratesapi.io/latest?base=USD', {
            method: 'GET'
        })
            .then(response => {

                return response.json();
            })
            .then(data => {

                this.setState({
                    usdToEuro: data.rates.EUR
                });

            });

    }

    render() {
        return (
            <CurrencyContext.Provider
                value={{
          currency: this.state.currency,
          toggleCurrency: this.toggleCurrency,
          usdToEuro : this.state.usdToEuro
        }}
            >
                {this.props.children}
            </CurrencyContext.Provider>
        )
    }
}

const CurrencyConsumer = CurrencyContext.Consumer;

export { CurrencyProvider, CurrencyConsumer, CurrencyContext }
