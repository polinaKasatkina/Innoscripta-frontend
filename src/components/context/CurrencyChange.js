import React from "react";
import { CurrencyConsumer } from './currencyContext';

class CurrencyChange extends React.Component {
    render() {

        return (
            <CurrencyConsumer>
                { context => (
                    <select onChange={context.toggleCurrency}>
                        <option value="USD">USD</option>
                        <option value="EUR">EUR</option>
                    </select>
                    )

                }

            </CurrencyConsumer>
        );
    }
}

export default CurrencyChange;
