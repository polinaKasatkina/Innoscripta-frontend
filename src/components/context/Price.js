import React, { Component } from "react";
import { CurrencyConsumer } from './currencyContext';

class Price extends Component {
    render() {

        return (
            <CurrencyConsumer>
                {({ currency, usdToEuro }) => (
                    <p className="menu-item-container--price your-order-item--price">{currency === "USD" ? "$" + this.props.price : "\u20AC" + (this.props.price * usdToEuro).toFixed(1)  }</p>
                )}
            </CurrencyConsumer>

        );
    }
}

export default Price;
