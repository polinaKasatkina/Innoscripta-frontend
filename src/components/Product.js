import React, { Component } from 'react';
import { Row, Col, Image } from 'react-bootstrap';
import { Link } from 'react-router-dom';

import Price from './context/Price';


import NumberSpinner from './NumberSpinner';

class Product extends Component {

    render() {

        let btnBlock = '';

        if (this.props.keyItem && this.props.keyItem === 4) {
            btnBlock = <Col md={4}>
                <div className="main-text">
                <p>Create your order right now</p>
                <Link className="btn btn-dark" to="/order">Order now!</Link>
                </div>
            </Col>
        }

        return (
            <React.Fragment>
                {btnBlock}
                <Col md={4}>
                    <div className="menu-item-container">
                        <Image src={'/images/' + this.props.product.name.toLowerCase().replace(/ /g, '-') + '.jpg'} fluid rounded/>
                        <p className="menu-item-container--name">{this.props.product.name}</p>
                        <p className="menu-item-container--desc">{this.props.product.description}</p>
                        <Row>
                            <Col md={6}>
                                <Price price={this.props.product.price} />
                            </Col>
                            <Col md={6} className="text-right">
                                <span className={"menu-item-container--category " + this.props.product.category.name.toLowerCase()}>{this.props.product.category.name}</span>
                            </Col>
                        </Row>
                        {
                            this.props.order
                            ? this.props.quantity === 0
                                ? <button className="btn btn-dark  btn-block btn-lg" onClick={() => this.props.addToCart(this.props.product)}>Add to
                                    Cart</button>
                                : <NumberSpinner quantity={this.props.quantity} updateCart={this.props.addToCart} product={this.props.product} />
                            : null
                        }
                    </div>
                </Col>
            </React.Fragment>
        )
    }

}

//Product.contextType = currencyContextConsumer;

export default Product;
