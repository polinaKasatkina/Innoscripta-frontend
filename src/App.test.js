import React from 'react';
import { render } from '@testing-library/react';
import Product from './components/Product';
import { CurrencyContext } from './components/context/currencyContext';

import renderer from 'react-test-renderer';

test('Product renders correctly on Main page', () => {

  const product = {
    id : 1,
    name : 'Margarita',
    description: 'cheese, tomatoes',
    price : 10,
    category :  { name : 'pizza' }
  };

  const tree = renderer
      .create(<CurrencyContext.Provider value={{ currency: "USD", usdToEuro : 1 }}><Product product={product} /></CurrencyContext.Provider>)
      .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Product renders correctly on Order page, not in cart', () => {

  const product = {
    id : 1,
    name : 'Margarita',
    description: 'cheese, tomatoes',
    price : 10,
    category :  { name : 'pizza' }
  };

  const tree = renderer
      .create(<CurrencyContext.Provider value={{ currency: "USD", usdToEuro : 1 }}>
        <Product product={product} order={true} addToCart={ () => {} } quantity={0} />
      </CurrencyContext.Provider>)
      .toJSON();
  expect(tree).toMatchSnapshot();
});


test('Product renders correctly on Order page, in cart', () => {

  const product = {
    id : 1,
    name : 'Margarita',
    description: 'cheese, tomatoes',
    price : 10,
    category :  { name : 'pizza' }
  };

  const tree = renderer
      .create(<CurrencyContext.Provider value={{ currency: "USD", usdToEuro : 1 }}>
        <Product product={product} order={true} addToCart={ () => {} } quantity={3} />
      </CurrencyContext.Provider>)
      .toJSON();
  expect(tree).toMatchSnapshot();
});
