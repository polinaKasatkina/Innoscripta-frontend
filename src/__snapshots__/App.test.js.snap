// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product renders correctly 1`] = `
Array [
  "",
  <div
    className="col-md-4"
  >
    <div
      className="menu-item-container"
    >
      <img
        className="img-fluid rounded"
        src="/images/margarita.jpg"
      />
      <p
        className="menu-item-container--name"
      >
        Margarita
      </p>
      <p
        className="menu-item-container--desc"
      >
        cheese, tomatoes
      </p>
      <div
        className="row"
      >
        <div
          className="col-md-6"
        >
          <p
            className="menu-item-container--price your-order-item--price"
          >
            $10
          </p>
        </div>
        <div
          className="text-right col-md-6"
        >
          <span
            className="menu-item-container--category pizza"
          >
            pizza
          </span>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Product renders correctly on Main page 1`] = `
Array [
  "",
  <div
    className="col-md-4"
  >
    <div
      className="menu-item-container"
    >
      <img
        className="img-fluid rounded"
        src="/images/margarita.jpg"
      />
      <p
        className="menu-item-container--name"
      >
        Margarita
      </p>
      <p
        className="menu-item-container--desc"
      >
        cheese, tomatoes
      </p>
      <div
        className="row"
      >
        <div
          className="col-md-6"
        >
          <p
            className="menu-item-container--price your-order-item--price"
          >
            $10
          </p>
        </div>
        <div
          className="text-right col-md-6"
        >
          <span
            className="menu-item-container--category pizza"
          >
            pizza
          </span>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Product renders correctly on Order page, in cart 1`] = `
Array [
  "",
  <div
    className="col-md-4"
  >
    <div
      className="menu-item-container"
    >
      <img
        className="img-fluid rounded"
        src="/images/margarita.jpg"
      />
      <p
        className="menu-item-container--name"
      >
        Margarita
      </p>
      <p
        className="menu-item-container--desc"
      >
        cheese, tomatoes
      </p>
      <div
        className="row"
      >
        <div
          className="col-md-6"
        >
          <p
            className="menu-item-container--price your-order-item--price"
          >
            $10
          </p>
        </div>
        <div
          className="text-right col-md-6"
        >
          <span
            className="menu-item-container--category pizza"
          >
            pizza
          </span>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-md-4"
        >
          Qty: 
        </div>
        <div
          className="col-md-8"
        >
          <div
            className="mb-3 input-group"
          >
            <div
              className="input-group-prepend"
            >
              <button
                className="btn btn-outline-secondary"
                data-dir="dwn"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                -
              </button>
            </div>
            <input
              className="text-center form-control"
              onChange={[Function]}
              type="text"
              value={3}
            />
            <div
              className="input-group-append"
            >
              <button
                className="btn btn-outline-secondary"
                data-dir="up"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                +
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Product renders correctly on Order page, not in cart 1`] = `
Array [
  "",
  <div
    className="col-md-4"
  >
    <div
      className="menu-item-container"
    >
      <img
        className="img-fluid rounded"
        src="/images/margarita.jpg"
      />
      <p
        className="menu-item-container--name"
      >
        Margarita
      </p>
      <p
        className="menu-item-container--desc"
      >
        cheese, tomatoes
      </p>
      <div
        className="row"
      >
        <div
          className="col-md-6"
        >
          <p
            className="menu-item-container--price your-order-item--price"
          >
            $10
          </p>
        </div>
        <div
          className="text-right col-md-6"
        >
          <span
            className="menu-item-container--category pizza"
          >
            pizza
          </span>
        </div>
      </div>
      <button
        className="btn btn-dark  btn-block btn-lg"
        onClick={[Function]}
      >
        Add to Cart
      </button>
    </div>
  </div>,
]
`;
